src/string.cr

```diff
@@ -2401,29 +2397,62 @@ class String
   #
   # "abcdef".compare("ABCDEF", case_insensitive: true) # => 0
   # "abcdef".compare("ABCDEG", case_insensitive: true) # => -1
+  #
+  # "heIIo".compare("heııo", case_insensitive: true, Unicode::CaseOptions::Turkic) # => 0
   # ```
-  def compare(other : String, case_insensitive = false)
+  def compare(other : String, case_insensitive = false, options = Unicode::CaseOptions::None)
     return self <=> other unless case_insensitive
 
-    reader1 = Char::Reader.new(self)
-    reader2 = Char::Reader.new(other)
-    ch1 = reader1.current_char
-    ch2 = reader2.current_char
+    if ascii_only? && other.ascii_only?
+      position = 0
+      byte1 = to_unsafe[0]
+      byte2 = other.to_unsafe[0]
+
+      while position < bytesize && position < other.bytesize
+        # Lowercase both bytes
+        if 65 <= byte1 <= 90
+          byte1 += 32
+        end
+        if 65 <= byte2 <= 90
+          byte2 += 32
+        end
 
-    while reader1.has_next? && reader2.has_next?
-      cmp = ch1.downcase <=> ch2.downcase
-      return cmp.sign if cmp != 0
+        comparison = byte1 <=> byte2
+        return comparison unless comparison == 0
 
-      ch1 = reader1.next_char
-      ch2 = reader2.next_char
-    end
+        position += 1
+        byte1 = to_unsafe[position]
+        byte2 = other.to_unsafe[position]
+      end
 
-    if reader1.has_next?
-      1
-    elsif reader2.has_next?
-      -1
+      if position < bytesize
```

Comment:
This could just be `(bytesize - other.bytesize).sign` which is shorter and probably even more efficient (not that it matters for this).

Reply:
Yes, it's a little bit faster, thanks!